pegasus/src/social_metadata.rb

```diff
@@ -301,6 +314,8 @@ def get_social_metadata_for_page(request)
       output["og:image:width"] = value[:width]
       output["og:image:height"] = value[:height]
       output["twitter:card"] = "photo"
+    when :image_twitter
+      output["twitter:image:src"] = "https://#{request.host}#{value[:path]}"
```

Comment:
Thank you for the output context! I'm now wondering does this require that we always list `image_twitter` after `image` so that it will override the `image` value? If so, would it be possible to have some kind of check in here to ensure that the `image_twitter` entry (if available) will override the `image` regardless of order? Like maybe line 313 could be `output["twitter:image:src"] = "https://#{request.host}#{value[:path]}" unless social_tag_set.includes(:image_twitter)`

Reply:
Good thinking! I tried `output["twitter:image:src"] = "https://#{request.host}#{value[:path]}" unless social_tag_set.include?(:image_twitter)` and that seems to work as expected.

Here's the /twitter-og-test page output that includes the `image_twitter` key:

<img width="819" alt="Screenshot 2023-10-10 at 8 41 15 AM" src="https://github.com/code-dot-org/code-dot-org/assets/9256643/d7adede1-1f64-4807-a19c-ed0bb8b7d06e">

And here's the /videos page output that only has the `image` key:

<img width="772" alt="Screenshot 2023-10-10 at 8 41 40 AM" src="https://github.com/code-dot-org/code-dot-org/assets/9256643/1734dae7-0410-49ec-be95-71eb6a200410">